resources/js/beatmap-discussions/discussion.tsx

```diff
@@ -62,62 +46,89 @@ function DiscussionTypeIcon({ type }: { type: DiscussionType | 'resolved' }) {
   );
 }
 
+function isBeatmapsetDiscussionJsonForShow(value: Props['discussion']): value is BeatmapsetDiscussionJsonForShow {
+  return 'posts' in value;
+}
```

Comment:
The parts that use it are more "render if component is on the page that uses this type (so just main discussions page)" rather than "render if this property is available", it's just coincidental that `posts` is currently enough to discriminate the type.

Reply:
I thought it's handled by the `readonly` props?